.github/workflows/inspektor-gadget.yml

```diff
@@ -78,100 +124,229 @@ jobs:
-      run: |
-        # Prevent releases with -dirty suffix due to forgotten entries in
-        # .gitignore.
-        changes="$(git status --porcelain)"
-        if [ -n "$changes" ] ; then
-          echo "$changes"
-          exit 1
-        fi
-
-        make kubectl-gadget-all
-
-        # Prepare assets for release and actions artifacts
-        for target in $(make list-kubectl-gadget-targets); do
-          platform=$(echo $target|cut -d- -f3-4)
-          mkdir $platform
-          cp kubectl-gadget-$platform $platform/kubectl-gadget
-          cp LICENSE $platform/
-          tar --sort=name --owner=root:0 --group=root:0 \
-            -czf inspektor-gadget-$platform.tar.gz -C $platform \
-            kubectl-gadget LICENSE
-          rm -rf $platform
-        done
-
-    - name: Build local-gadget for all platforms
-      run: |
-        make local-gadget-all
-
-        # Prepare assets for release and actions artifacts
-        for target in $(make list-local-gadget-targets); do
-          platform=$(echo $target|cut -d- -f3-4)
-          mkdir $platform
-          cp local-gadget-$platform $platform/local-gadget
-          cp LICENSE $platform/
-          tar --sort=name --owner=root:0 --group=root:0 \
-            -czf local-gadget-$platform.tar.gz -C $platform \
-            local-gadget LICENSE
-          rm -rf $platform
-        done
-
+  test-basic:
+    name: Basic unit tests
+    # level: 0
+    runs-on: ubuntu-latest
+    steps:
+    - name: Setup go 1.17
+      uses: actions/setup-go@v1
+      with:
+        go-version: 1.17
+      id: go
+    - name: Cache go 1.17
+      uses: actions/cache@v2
+      with:
+        path: |
+          ~/.cache/go-build
+          ~/go/pkg/mod
+        key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+        restore-keys: |
+          ${{ runner.os }}-go-
+    - name: Check out code
+      uses: actions/checkout@v1
+    - name: Cache debian packages
+      uses: actions/cache@v2
+      id: cache-debs
+      with:
+        path: "~/cache-debs"
+        key: cache-debs-libbpf-1:0.4.0-1ubuntu1-libseccomp-2.5.1-1ubuntu1~20.04.1
+    - name: Install debian packages
+      uses: ./.github/actions/install-debian-packages
+      with:
+        cache-hit: ${{steps.cache-debs.outputs.cache-hit}}
     - name: Basic unit tests
       run: |
         make test
 
+  test-controller:
+    name: Unit tests for the controller
+    # level: 0
+    runs-on: ubuntu-latest
+    steps:
+    - name: Setup go 1.17
+      uses: actions/setup-go@v1
+      with:
+        go-version: 1.17
+      id: go
+    - name: Cache go 1.17
+      uses: actions/cache@v2
+      with:
+        path: |
+          ~/.cache/go-build
+          ~/go/pkg/mod
+        key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+        restore-keys: |
+          ${{ runner.os }}-go-
+    - name: Check out code
+      uses: actions/checkout@v1
     - name: Unit tests for the controller
       run: |
         make controller-tests
```

Comment:
IMO we can combine this with the test-basic job. Each one is taking like 1 minute, so they won't be the bottleneck.